std/libstd/Cargo.toml

```diff
@@ -16,13 +16,13 @@ crate-type = ["dylib", "rlib"]
 [dependencies]
 alloc = { path = "../liballoc" }
 cfg-if = { version = "0.1.8", features = ['rustc-dep-of-std'] }
-panic_unwind = { path = "../libpanic_unwind", optional = true }
-panic_abort = { path = "../libpanic_abort" }
+panic_unwind = { path = "../../src/libpanic_unwind", optional = true }
+panic_abort = { path = "../../src/libpanic_abort" }
```

Comment:
In terms of which crates to move where:

* Can we make sure that crates from `/std` depend only on other crates in std? IE, if we just eagerly move every `../../src` dependecy to `../../std`, would the split still be meaningful? It seems like it should be the case (because runtime code can depend only on runtime code). 
* If this is feasible, can we add a tidy check to ensure that `std -> src` dependencies don't resurface?
* Would the same hold in the opposite direction? Can we make sure that crates from `/src` don't depend on crates from `/std`? Can we enforce that? 

Ie, it seems like src and std *should* be independent bodies of code (one runs on target, another on host, etc), and, if that is indeed the case, I'd like this to be very cleary enforced, as "module X does **not** depend on module Y" is the strongest property you can get in terms of software architecture at large.

Reply:
I agree that `/src` and `/std` should be entirely split, possibly modulo build dependencies (build_helper is "special"). That may mean e.g. a separate directory like `/build_system` (or `/rustbuild` perhaps) that anything can depend on, not sure.